client/core/core.go

```diff
@@ -2356,16 +2363,52 @@ func (c *Core) DiscoverAccount(dexAddr string, appPW []byte, certI interface{})
 		return dc.exchangeInfo(), false, nil
 	}
 
-	incompleteReg, paid, err := c.registerHD(dc, crypter)
+	// Setup our account keys and attempt to authorize with the DEX.
+	creds := c.creds()
+	encKey, _, err := dc.acct.setupCryptoV2(creds, crypter)
 	if err != nil {
-		return nil, false, err
+		return dc.exchangeInfo(), false, newError(acctKeyErr, "setupCryptoV2 error: %v", err)
+	}
+
+	// Instead of calling register, attempt authDEX.
+	err = c.authDEX(dc)
```

Comment:
One difference with this approach is that `err == nil` when the account is suspended (primarily for purposes of negotiating active swaps).
Connect response has `score`, but client does not know ban score, which is the threshold to determine suspension.
Can the user manually disable the account in the /settings page?  I haven't tried this for accounts derived from the hd keys, only legacy.

Reply:
Just pushed a commit to attempt to address this https://github.com/decred/dcrdex/pull/1201/commits/44d208740bca3139040fcfbe3f8d9e8400e91d36

```
[INF] CORE: HD account key for 127.0.0.1:17273 was reported as suspended. Register to make a new one.
"POST http://127.0.0.3:5758/api/discoveracct HTTP/1.1" from 127.0.0.1:40232 - 200 968B in 1.247533373s
```

takes to Confirm screen with the markets table, then click confirm for it to call Register, which discovers the HD key won't work and makes a legacy account...

```
[INF] CORE: HD account key for 127.0.0.1:17273 was reported as suspended.
[INF] CORE: Generating a legacy-style key to register with 127.0.0.1:17273.
```

While this is the current behavior on master, it does make me a little concerned that the user might not realize it went the "legacy" account path here, meaning the account cannot be recovered by the seed, only export/import.  I think this just means we probably do want to pursue multiple child keys at the derivation path for a given DEX host.